robottelo/hosts.py

```diff
@@ -1776,6 +1787,50 @@ def _swap_nailgun(self, new_version):
         to_clear = [k for k in sys.modules if 'nailgun' in k]
         [sys.modules.pop(k) for k in to_clear]
 
+    def enable_ipv6_http_proxy(self):
+        # Execute procedures for enabling IPv6 HTTP Proxy
+        if all[settings.server.is_ipv6, settings.server.http_proxy_ipv6_url]:
+            proxy_name = 'Robottelo IPv6 Automation Proxy'
+            if self.cli.HttpProxy.exists(search=('name', proxy_name)):
+                http_proxy = self.cli.HttpProxy.create(
+                    {
+                        'name': proxy_name,
+                        'url': settings.server.http_proxy_ipv6_url,
+                    }
+                )
+                self.cli.Settings.set(
+                    {
+                        'name': 'content_default_http_proxy',
+                        'value': proxy_name,
+                    }
+                )
+                return http_proxy
+            else:
+                logger.info(
+                    'The IPv6 HTTP Proxy is already enabled!! Skipping the IPv6 HTTP Proxy setup.'
+                )
+                http_proxy = self.cli.HttpProxy.list({'search': f'name={proxy_name}'})
+                if http_proxy:
+                    return http_proxy[0]
+                return None
+        else:
+            logger.warning(
+                'The IPv6 HTTP Proxy setting is not enabled!! Skipping the IPv6 HTTP Proxy setup.'
+            )
+            return None
+
+    def disable_ipv6_http_proxy(self, http_proxy):
```

Comment:
Is there a scenario when a Satellite can have more than one HTTP proxy? If not, could we have the proxy as a Satellite class parameter? We would not need to specify what HTTP proxy to disable.

Reply:
Agree. But I am not sure if we could have more than one or not. I would like Teams to respond on that request.

Ping @vsedmik

Reply:
In Satellite you can have defined more HTTP proxy entities. But for communication you can use just one. I mean one for Foreman (http_proxy) and one for Katello (content_default_http_proxy). I believe we need to set both of them in settings to route all traffic from Satellite through them.

Reply:
@vsedmik So can we create one http proxy (as we are doing here) and set it for both foreman and katello? Do we have different settings for both of them which we can set ?

Reply:
Correct, create one HTTP proxy and set it Settings in `http_proxy` and `content_default_http_proxy`. I'm not aware of any other similar settings we should set.

Reply:
Thanks @vsedmik , I can fix that right away.

Reply:
Fixed!